### PR TITLE
Allow branding to specify SMTP settings to send emails

### PR DESCRIPTION
**Pull request by norkans7**
Fixes https://textit.sentry.io/issues/4493889128/

### CODE REVIEW

File: temba/utils/email.py [hunk 1]
```diff
@@ -114,7 +114,25 @@ def send_template_email(recipients, subject, template, context, branding):
...
+
+        connection = get_smtp_connection(
+            None,
+            fail_silently=False,
+            host=smtp_host,
+            port=smtp_port,
+            username=smtp_username,
+            password=smtp_password,
+            use_tls=smtp_tls,
+        )
```
Comment (rowanseymour) [inline temba/utils/email.py:133]: hmm maybe we should say that the smtp settings on the parent org become the defaults? @ericnewcomer

Comment (ericnewcomer) [reply temba/utils/email.py:133]: Ya, was SMTP ever a branding config? I thought this was just for orgs. I like the idea of using the parent org as the default unless it is overridden.

Comment (rowanseymour) [reply temba/utils/email.py:133]: Although... until now we've said those SMTP settings are for flows.. and you still get textit emails for things like exports.. does it make sense to mix those?

Comment (ericnewcomer) [reply temba/utils/email.py:133]: Hmmm.. that is a good point. I guess the difference here is one is potentially contact-facing and the other is user-facing. Would it be terrible to use it for both? Password resets is the only one that seems maybe kinda weird.

Comment (ericnewcomer) [reply temba/utils/email.py:133]: Ok, chatted with @rowanseymour a bit on this and we agreed that this does indeed make the most sense to be part of BRAND. That said, let's change this to be under the same parent key of `smtp`. So it should look something like..

```
BRAND = {
   ...
   "smtp": {
      "from": ...
      "host": ...
      "port": ...
      ...
   }
}
```

We should add defaults to BRAND so we don't need those in code.

Comment (norkans7) [reply temba/utils/email.py:133]: Okay, I make the changes

File: temba/settings_common.py [hunk 2]
```diff
@@ -282,6 +282,14 @@
     "domain": "app.rapidpro.io",
     "styles": [],
     "email": "join@rapidpro.io",
+    "from_email": DEFAULT_FROM_EMAIL,
```
[the remark this answers is not in the document]
Comment (norkans7) [reply temba/settings_common.py:285]: We had `from_email` key already in there but I can move that

**Comment by rowanseymour** [comment]
Does this work if a brand is only specifying `from_email` and not the other SMTP settings? Do we need to go update legacy configs in the database? @ericnewcomer

**Comment by ericnewcomer** [comment]
If you share the format for the branding updates, I can take care of those. Let's get this merged.

**Comment by norkans7** [comment]
@ericnewcomer we need to add `smtp` key as 

```
smtp={
                "host": "smtp.example.com",
                "port": 587,
                "username": "foo",
                "password": "bar",
                "use_tls": True,
                "from_email": "support@mybrand.com",
            }
```

**Pull request closed without merging**